### DISCUSSION TITLE
Resource group is not working as expect

### DISCUSSION THREAD

**Discussion by njalan** [trinodb/trino#23266] [General]
Below is my resource group file. 
Sub group **opn** is for report which is running slow and it takes 30 seconds by 5 users.
Sub group dss is for report which is running fase and it takes 2 seconds by 100 users. Concurrency is very high.
Report running by subgroup opn is more import then report running by group **dss**.
My point is that I want to make sure Sub group **opn* at any time has 20% of the resource but actually when there are high workload for Sub group dss then **opn* is also affected.
How can I make sure Sub group opn has the high priority then dss and make sure it always has 20% of the resource. I just want to make sure the reports running by sub group opn always get enough resource to finish. 

{
  "rootGroups": [
    {
      "name": "global",
      "softMemoryLimit": "90%",
      "hardConcurrencyLimit": 80,
      "maxQueued": 300,
      "jmxExport": true,
      "subGroups": [
        {
          "name": "zlb",
          "softMemoryLimit": "60%",
          "hardConcurrencyLimit": 20,
          "maxQueued":100
        },
       {
          "name": "opn",
          "softMemoryLimit": "20%",
          "hardConcurrencyLimit":20,
          "maxQueued": 40
        },
        {
          "name": "dss",
          "softMemoryLimit": "20%",
          "hardConcurrencyLimit":30,
          "maxQueued": 160
        }
      ]
    }
  ],
  "selectors": [

    {
      "user": "xxxx",
      "group": "global.opn"
    },
    {
      "user": "yyy",
      "group": "global.dss"
    },

    {
      "user": ".*",
      "group": "global.zlb"
    }

  ],
  "cpuQuotaPeriod": "1h"
}

**Comment by hashhar** [comment]
Resource groups as they exist today only provide admission control. Once a query is selected for execution by resource groups they no longer have any control i.e. they cannot limit amount of resources a group can use at runtime, they can only prevent future queries to run in a group if a group has exceeded it's limits.
